### ISSUE TITLE
Custom taxonomy throwing an error

### ISSUE THREAD

**Issue by hellor0bot** [WordPress/gutenberg#4455]
I'm registering a custom taxonomy like this:

```
/**
 * Register custom featured taxonomy
 */
function prefix_register_featured_taxonomy() {

	$labels = array(
		'name' => esc_html__( 'Featured Locations', 'text-domain' ),
		'singular_name' => esc_html__( 'Featured Location', 'text-domain' ),
		'all_items' => esc_html__( 'All Locations', 'text-domain' ),
	);

	$args = array(
		'label' => esc_html__( 'Featured Locations', 'text-domain' ),
		'labels' => apply_filters( 'prefix_featured_taxonomy_labels', $labels ),
		'public' => false,
		'hierarchical' => true,
		'show_ui' => true,
		'show_in_rest' => true,
		'show_in_menu' => false,
		'show_in_nav_menus' => false,
		'query_var' => true,
		'rewrite' => false,
		'show_admin_column' => true,
		'show_in_quick_edit' => true,
		'sort' => true,
		'capabilities' => array(
			'manage_terms' => false,
			'edit_terms' => false,
			'delete_terms' => false,
		),
	);

	register_taxonomy( 'prefix_post_featured', array( 'post' ), apply_filters( 'prefix_featured_taxonomy_args', $args ) );
}

add_action( 'init', 'prefix_register_featured_taxonomy', 10 );
```
Then I'm adding terms manually like this:

```
/**
 * Add default featured terms upon theme activation
 */
function prefix_add_featured_terms() {

	if ( get_option( 'prefix_featured_terms_added' ) ) {

		// Return if terms have already been added.
		return;

	} else {

		// Array of Featured Locations.
		$featured_terms = array(
			'slider'   => esc_html__( 'Post Slider', 'text-domain' ),
			'tiles'    => esc_html__( 'Post Tiles', 'text-domain' ),
			'carousel' => esc_html__( 'Post Carousel', 'text-domain' ),
			'archive'  => esc_html__( 'Post Archive', 'text-domain' ),
			'widget'   => esc_html__( 'Posts Widget', 'text-domain' ),
		);

		// Add terms to custom taxonomy Featured Locations.
		foreach ( apply_filters( 'prefix_featured_terms', $featured_terms ) as $term => $name ) {
			if ( ! term_exists( $name, 'prefix_post_featured' ) ) {
				wp_insert_term(
					$name, 'prefix_post_featured', $args = array(
						'slug' => $term,
					)
				);
			}
		}

		// Set an option, that terms have been added.
		update_option( 'prefix_featured_terms_added', true );

	}

}

add_action( 'init', 'prefix_add_featured_terms', 20 );
```

The taxonomy terms aren't displayed in the Gutenberg sidebar and there's an error:

`Warning: call_user_func() expects parameter 1 to be a valid callback, function 'post_categories_meta_box' not found or invalid function name in /Users/user/Dev/projects/stage/wp-content/plugins/gutenberg/lib/meta-box-partial-page.php on line 256`

**Comment by bobbingwide** [comment]
I got a similar message with `post_tags_meta_box` when associating custom tags to the `post` post type.

**Comment by bobbingwide** [comment]
I've updated my own issue with details of the fixes that I have applied locally to make it work. I believe that Gutenberg should attempt to address both these fixes, otherwise it will break backward compatibility. 
1. I changed my custom taxonomy to support `show_in_rest`.
2. I added a test in `gutenberg_filter_meta_boxes`. I didn't replace the test since I didn't know if the filter was invoked elsewhere when `callback` is not `gutenerg_override_meta_box_callback`.
See http://github.com/bobbingwide/oik-a2z/issues/8
